system-monitor/MonitorThread.py

```diff
@@ -0,0 +1,72 @@
+import threading
+import time
+import os
+from datetime import datetime
+from subprocess import Popen, PIPE
+
+class MonitorThread(threading.Thread):
+
+    def __init__(self, *args, **kwargs):
+        super(MonitorThread, self).__init__(*args, **kwargs)
+        self._stop = threading.Event()
+
+    def stop(self):
+        print('TH | Stopping thread!')
+        self._stop.set()
+
+    def stopped(self):
+        return self._stop.isSet()
+
+    def set_pids(self, pids):
+        self.pids = pids
+    
+    def set_function_name(self, name):
+        self.function_name = name
+
+    def run(self):
+        top_cmd = ['top','-b','-n 1', '-p', self.pids]
+        tail_cmd = ['tail', '-n+8']
+        awk_cmd = ['awk', '{print $1 "\t" $9 "\t" $10}']
+        
+        date_time = datetime.now()
+        experiment_datetime = date_time.strftime("%d_%m_%Y_%H_%M_%S")
+        path = "experiments/"+ self.function_name + "_" + experiment_datetime
```

Comment:
is `+ "_" + experiment_datetime` really needed? Do we want to name directories like that? IMHO it would be clearer to suffix with number of experimsent, but I wouldn't care about this now. Firstly, we want to run all functions once